Assets/MixedRealityToolkit/Inspectors/Profiles/MixedRealityProfileCloneWindow.cs

```diff
@@ -0,0 +1,320 @@
+            childProfileAssetName = "New " + childProfileTypeName;
+
+            // Find all the serialized properties for sub-profiles
+            SerializedProperty iterator = childSerializedObject.GetIterator();
+            System.Type basePropertyType = typeof(BaseMixedRealityProfile);
+
+            while (iterator.Next(true))
+            {
+                SerializedProperty subProfileProperty = childSerializedObject.FindProperty(iterator.name);
+
+                if (subProfileProperty == null)
+                    continue;
+
+                if (!subProfileProperty.type.Contains("PPtr<$")) // Not an object reference type
+                    continue;
+
+                string subProfileTypeName = subProfileProperty.type.Replace("PPtr<$", string.Empty).Replace(">", string.Empty).Trim();                
+                System.Type subProfileType = FindProfileType(subProfileTypeName);
+                if (subProfileType == null)
+                    continue;
+
+                if (!basePropertyType.IsAssignableFrom(subProfileType))
+                    continue;
+
+                subProfileActions.Add(new SubProfileAction(
+                    ProfileCloneBehavior.UseExisting,
+                    subProfileProperty,
+                    subProfileProperty.objectReferenceValue,
+                    subProfileType));
+            }
+
+            Vector2 minWindowSize = MinWindowSizeBasic;
+            minWindowSize.y = Mathf.Max(minWindowSize.y, subProfileActions.Count * SubProfileSizeMultiplier);            
+            cloneWindow.minSize = minWindowSize;
+
+            // If there are no sub profiles, limit the max so the window isn't spawned too large
+            if (subProfileActions.Count <= 0)
+                cloneWindow.maxSize = minWindowSize;
+        }
+
+        private void OnGUI()
+        {
+            if (cloneWindow == null || parentProfile == null || childProfile == null)
+            {
+                Close();
+                return;
+            }
+
+            EditorGUILayout.BeginVertical(EditorStyles.helpBox);
+            EditorGUILayout.ObjectField("Cloning profile", childProfile, typeof(BaseMixedRealityProfile), false);
+            EditorGUILayout.ObjectField("from parent profile", parentProfile, typeof(BaseMixedRealityProfile), false);
+            EditorGUILayout.EndVertical();
+            EditorGUILayout.Space();
+
+            if (subProfileActions.Count > 0)
+            {
+                EditorGUILayout.HelpBox("This profile has sub-profiles. By defult your clone will reference the existing profiles. If you want to specify a different profile, or if you want to clone the sub-profile, use the options below.", MessageType.Info);
+
+                EditorGUILayout.BeginVertical();
+
+                for (int i = 0; i < subProfileActions.Count; i++)
+                {
+                    GUI.color = Color.white;
```

Comment:
nit: may be worth defining some specific visual values like this as readonly/consts or at least fields in the class to that they can be consistenlty reused/updated in one location